### DISCUSSION TITLE
Caching Tile-Data with no content

### DISCUSSION THREAD

**Discussion by marq24** [mapsforge/vtm#922] [Ideas]
Let me start with that I solved the issue in my local application already - now I want to ask, if you are interested in a PR.

Here is the situation: Some tile servers (like mapbox) will send you HTTP:200 but with an empty image [ContentLength: 0] when requesting none existing tiles - typically this happens, when you are at the north/south border.

While the complete system handles this situation quite well, I don't like the fact, that when I use a TileCache with my BitmapLayer, that the ZeroByte Tiles will be requested again and again. So even if this request will be handled fast & reliable - they are obsolete.

In my local implementation the TileCache will store in such a case byte[]{-1} for the requested tile and in return the UrlTileDataSource query method will call simply `sink.completed(TILE_NOT_FOUND);` and return when the TileReader indicated, that the content was just a byte[] of length 1 (with the value -1).

The change includes interface changes in `HttpEngine` and `ITileCache` which might make this a bit messy.

just let me know, if you are interested.

**Comment by devemux86** [comment]
Thanks for the suggestion.

Although there does not seem to be any problem in the [BitmapTileActivity](https://github.com/mapsforge/vtm/blob/master/vtm-android-example/src/org/oscim/android/test/BitmapTileActivity.java) sample.

> The change includes interface changes in HttpEngine and ITileCache which might make this a bit messy.

Small (or edge) cases can not justify major interface changes (and maintenance).
We could continue with the existing code, especially since there are no problems.
